fs/info.py

```diff
@@ -317,9 +317,12 @@ def modified(self):
                 namespace is not in the Info.
 
         """
-        self._require_namespace("details")
-        _time = self._make_datetime(self.get("details", "modified"))
-        return _time
+        try:
+            self._require_namespace("details")
+            return self._make_datetime(self.get("details", "modified"))
+        except MissingInfoNamespace:
+            self._require_namespace("modified")
+            return self._make_datetime(self.get("modified", "modified"))
```

Comment:
Since the `MissingInfoNamespace` is actually raised by `self._require_namespace(...)`, I think it would be nicer to do the following (minimal branching, and at most one exception raised):

```python
@property
def modified(self):
    namespace =  "modified" if "modified" in self.raw else "details"
    self._require_namespace(namespace)
    return self._make_datetime(self.get(namespace, "modified"))
```

Reply:
I don't feel like that's a good change just to get rid of one possible `raise`. That would make `modified` the only property in `Info` to not use `get` but access `raw` instead, reducing encapsulation of that class.

Reply:
Then I guess using `has_namespace` instead, which is a public method, would be fine:
```python
@property
def modified(self):
    namespace =  "modified" if self.has_namespace("modified") else "details"
    self._require_namespace(namespace)
    return self._make_datetime(self.get(namespace, "modified"))
```

Reply:
Done.